placement/algo/sharded_helper.go

```diff
@@ -738,3 +738,80 @@ func newShards(shardIDs []uint32) []shard.Shard {
 	}
 	return r
 }
+
+func removeInstanceFromList(list []placement.Instance, instanceID string) []placement.Instance {
+	for i, instance := range list {
+		if instance.ID() == instanceID {
+			last := len(list) - 1
+			list[i], list[last] = list[last], list[i]
```

Comment:
nit: do we need the swap here? Seems it might be simpler to do:

```
last := len(list)-1
list[i] = list[last]
return list[:last]
```

Reply:
yeah, could save the swap